### PR TITLE
[Documentation][Cookbook] Disabling order confirmation email

### LINKED ISSUE #7763: Disabling order confirmation email
On beta1, emails were sent using the state machine, which wasn't perfect, but it allowed me to disable the order confirmation email, and send out my own payment confirmation email.

Currently there are a few issues:

- I initially thought it was the event `sylius.order.post_payment`, but its not. It should be renamed `sylius.order.post_select_payment`
- The event that it is supposed to be (`sylius.payment.post_complete`) doesn't fire, but there is already a ticket open.
- I can't seem to easily disable the current email sent via listening to `sylius.order.post_complete`.

Finally, I just wanted to ask, why does Sylius send order confirmation emails before payment anyways? I would think that most people would want an order confirmation email once something has been paid for.

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no |
| New docs?    | yes |
| Related tickets | fixes #7763 |
| License         | MIT |

### CODE REVIEW

**Comment by NeverResponse** [comment]
Pretty sure you can get the same end result with:
``` yaml
sylius_mailer:
    emails:
        order_confirmation:
            enabled: false
```

**Comment by inssein** [comment]
@NeverResponse ah totally missed that. That said, I actually do what is suggested in this doc because I send the same email after payment (through the state machine).

File: docs/cookbook/disabling-order-confirmation-email.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        public function build(ContainerBuilder $container)
+        {
+            parent::build($container);
+
+            $container->addCompilerPass(new MailPass());
+        }
+    }
+
+That's it, we have removed the definition of the listnere that is responsible for sending the order confirmation email.
```
Comment (NeverResponse) [inline docs/cookbook/disabling-order-confirmation-email.rst]: `listnere` => `listener`

**Comment by pjedrzejewski** [comment]
👍

**Pull request merged**